packages/tendermint-rpc/src/client.spec.ts

```diff
@@ -124,118 +129,217 @@ function defaultTestSuite(rpcFactory: () => RpcClient, adaptor: Adaptor): void {
+      const client = new Client(rpcFactory(), adaptor);
 
-    const find = randomString();
-    const me = randomString();
-    const tx = buildKvTx(find, me);
-
-    const txRes = await client.broadcastTxCommit({ tx: tx });
-    expect(responses.broadcastTxCommitSuccess(txRes)).toEqual(true);
-    expect(txRes.height).toBeTruthy();
-    const height: number = txRes.height || 0; // || 0 for type system
-    expect(txRes.hash.length).not.toEqual(0);
-    const hash = txRes.hash;
-
-    await tendermintSearchIndexUpdated();
-
-    // find by hash - does it match?
-    const r = await client.tx({ hash: hash, prove: true });
-    // both values come from rpc, so same type (Buffer/Uint8Array)
-    expect(r.hash).toEqual(hash);
-    // force the type when comparing to locally generated value
-    expect(r.tx).toEqual(tx);
-    expect(r.height).toEqual(height);
-    expect(r.proof).toBeTruthy();
-
-    // txSearch - you must enable the indexer when running
-    // tendermint, else you get empty results
-    const query = buildQuery({ tags: [{ key: "app.key", value: find }] });
-
-    const s = await client.txSearch({ query: query, page: 1, per_page: 30 });
-    // should find the tx
-    expect(s.totalCount).toEqual(1);
-    // should return same info as querying directly,
-    // except without the proof
-    expect(s.txs[0]).toEqual({ ...r, proof: undefined });
-
-    // ensure txSearchAll works as well
-    const sall = await client.txSearchAll({ query: query });
-    // should find the tx
-    expect(sall.totalCount).toEqual(1);
-    // should return same info as querying directly,
-    // except without the proof
-    expect(sall.txs[0]).toEqual({ ...r, proof: undefined });
-
-    // and let's query the block itself to see this transaction
-    const block = await client.block(height);
-    expect(block.block.txs.length).toEqual(1);
-    expect(block.block.txs[0]).toEqual(tx);
+      // Run in parallel to increase chance there is no block between the calls
+      const [status, blockchain] = await Promise.all([client.status(), client.blockchain()]);
+      const height = status.syncInfo.latestBlockHeight;
 
-    client.disconnect();
-  });
+      expect(blockchain.lastHeight).toEqual(height);
+      expect(blockchain.blockMetas.length).toBeGreaterThanOrEqual(3);
+      expect(blockchain.blockMetas[0].header.height).toEqual(height);
+      expect(blockchain.blockMetas[1].header.height).toEqual(height - 1);
+      expect(blockchain.blockMetas[2].header.height).toEqual(height - 2);
 
-  it("can paginate over txSearch results", async () => {
-    pendingWithoutTendermint();
-    const client = new Client(rpcFactory(), adaptor);
+      client.disconnect();
+    });
+
+    it("can limit by maxHeight", async () => {
+      pendingWithoutTendermint();
+      const client = new Client(rpcFactory(), adaptor);
+
+      const height = (await client.status()).syncInfo.latestBlockHeight;
+      const blockchain = await client.blockchain(undefined, height - 1);
```

Comment:
Ah, this will get any sequence of blocks. And from the future?

Reply:
> I would ask for a test that given status queries blockchain(height, height+1).

I tested this in my proof code. It throws an error like:

`Error: {"code":-32603,"message":"Internal error","data":"min height 1005 can't be greater than max height 1004"}`

Reply:
I'll test future blocks. But I'm pretty sure this is query+filtering.

The min > max case should return an empty result instead of an error in my opinion. But the error is more helpful for debugging since this is not intended usually.

Reply:
The error was not a `min > max` case (which can throw error), but `max > current height` case

Reply:
`max > current height` is not the problem (added a test here). But I found the problem: `min > current height`

Reply:
In #395 I worked around it using

```ts
      // start from current height to avoid backend error for minHeight in the future
      let header = (await this.tmClient.blockchain(height, searchHeight)).blockMetas
        .map((meta) => meta.header)
        .find((h) => h.height === searchHeight);
```